examples/type-level-boolean-programming

```diff
@@ -0,0 +1,38 @@
+/// A list that tracks whether its length is even or odd.
+enum EOList[a: Type, even: Bool] {
+    case EONil
+    case EOCons(a, EOList[a, not even])
+}
+
+/// Returns an empty and even list.
+def mkNilEO(): EOList[a, true] = EOList.EONil
```

Comment:
Can we drop all the EO suffixes? I think we should only use them for the EOType.